### DISCUSSION TITLE
Is there a way to search for matches in the forehead part of a string ?

### DISCUSSION THREAD

**Discussion by nemovalex** [TarekRaafat/autoComplete.js#291] [Q&A]
Hello everyone and thanks to authors of this JS library. For instance, in my data I have an array like this 
`
data: ['How to learn HTML', 'Hello amigo',' I need burgers']
`
If I write 'learn' the autocomplete library finds 'How to learn HTML', it is OK. However if I write something else BEFORE keywords, like 'best ways to learn HTML' the plugin does not find matching. Is there any possible way to make this library find matching in any part of string ?

**Comment by folknor** [comment]
Yes, you need to use a custom `searchEngine`: https://tarekraafat.github.io/autoComplete.js/#/configuration?id=searchengine-optional

But setting `searchEngine: "loose"` might be enough as well.

Possibly useful to read this as well https://github.com/TarekRaafat/autoComplete.js/issues/268

**Reply by nemovalex** [reply]
Thanks for the answer, but it does not help. If I have 'plumbing work' in Array and write in input something like 'need a plumber' I want to get 'plumbing work` as I have coincidence - 'plumb'.  However, if I start to write with 'plumb....' it will work, but if I start to write with another word like 'I need', 'need', 'looking for...plumber' for example , because people usually write something like this ,it does not work. I guess it is not possible (?). Anyway, thank you, and sorry for my English...

**Reply by Mark-Hetherington** [reply]
You can write your own search engine, with custom matching logic. As noted by the link Folknor provided you will need to provide a function that takes two parameters. e.g.
```
searchEngine:(query, record) => {
  if (customMatchLogic(query, record)) return record;
}
```
The loose matching that is built in requires the query characters typed to be found in the record in the same order. It isn't capable of selectively matching part of the string. You might be able to use something like https://www.npmjs.com/package/fuse.js as your search engine.
